pkg/cmd/operator/cleanupjob.go

```diff
@@ -0,0 +1,134 @@
+	apierrors "k8s.io/apimachinery/pkg/util/errors"
+	cliflag "k8s.io/component-base/cli/flag"
+	"k8s.io/klog/v2"
+)
+
+type CleanupJobOptions struct {
+	AuthToken   string
+	NodeAddress string
+
+	scyllaClient *scyllaclient.Client
+}
+
+func NewCleanupJobOptions(streams genericclioptions.IOStreams) *CleanupJobOptions {
+	return &CleanupJobOptions{}
+}
+
+func NewCleanupJobCmd(streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewCleanupJobOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:   "cleanup-job",
+		Short: "Runs a cleanup procedure against a node.",
+		Long:  "Runs a cleanup procedure against a node.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate()
+			if err != nil {
+				return err
+			}
+
+			err = o.Complete()
+			if err != nil {
+				return err
+			}
+
+			err = o.Run(streams, cmd)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	cmd.Flags().StringVarP(&o.AuthToken, "auth-token", "", o.AuthToken, "Path to a file containing auth token.")
+	cmd.Flags().StringVarP(&o.NodeAddress, "node-address", "", o.NodeAddress, "Address of a node where cleanup will be performed.")
+
+	return cmd
+}
+
+func (o *CleanupJobOptions) Validate() error {
+	var errs []error
+
+	if len(o.AuthToken) == 0 {
+		errs = append(errs, fmt.Errorf("auth-token cannot be empty"))
+	}
+
+	if len(o.NodeAddress) == 0 {
+		errs = append(errs, fmt.Errorf("node-address cannot be empty"))
+	}
+
+	return apierrors.NewAggregate(errs)
+}
+
+func (o *CleanupJobOptions) Complete() error {
+	var err error
+
+	buf, err := os.ReadFile(o.AuthToken)
+	if err != nil {
+		return fmt.Errorf("can't read auth token file at %q: %w", o.AuthToken, err)
+	}
+
+	authToken, err := helpers.ParseTokenFromConfig(buf)
+	if err != nil {
+		return fmt.Errorf("can't parse auth token file at %q: %w", o.AuthToken, err)
+	}
+
+	o.scyllaClient, err = controllerhelpers.NewScyllaClientFromToken([]string{o.NodeAddress}, authToken)
+	if err != nil {
+		return fmt.Errorf("can't create scylla client: %w", err)
+	}
+
+	return nil
+}
+
+func (o *CleanupJobOptions) Run(streams genericclioptions.IOStreams, cmd *cobra.Command) error {
+	klog.InfoS("Starting the cleanup job", "version", version.Get())
+	cliflag.PrintFlags(cmd.Flags())
+
+	stopCh := signals.StopChannel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go func() {
+		<-stopCh
+		cancel()
+	}()
+
+	keyspaces, err := o.scyllaClient.Keyspaces(ctx)
+	if err != nil {
+		return fmt.Errorf("can't get list of keyspaces: %w", err)
+	}
+
+	klog.InfoS("Discovered keyspaces for cleanup", "keyspaces", keyspaces)
+
+	for _, keyspace := range keyspaces {
+		klog.InfoS("Starting a keyspace cleanup", "keyspace", keyspace)
+		err = o.scyllaClient.Cleanup(ctx, o.NodeAddress, keyspace)
+		if err != nil {
+			return fmt.Errorf("can't cleanup keyspace %q: %w", keyspace, err)
+		}
+	}
+
+	klog.InfoS("Cleanup completed")
```

Comment:
could be useful to trace the duration, possibly on a keyspace level with higher loglevel

Reply:
done